exercises/practice/paasio/src/test/java/PaasioTest.java

```diff
@@ -0,0 +1,337 @@
+    void checkIfNullPointerExceptionIsThrownWhenNullIsPassedInsteadOfByteArray() {
+
+        assertThatThrownBy(() -> {
+            Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream);
+            fileOperations.read(null);
+        }).isInstanceOf(NullPointerException.class);
+
+    }
+
+    @Test
+    void checkIfDataFollowsTheOffsetWhileWritingInByteArray() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            byte[] initialMessage = "Hello! ".getBytes();
+            byte[] messageArray = Arrays.copyOf(initialMessage, 50);
+
+            int length = messageArray.length - initialMessage.length - 2;
+            //It will read data into the messageArray and will begin writing it from its  initialMessage.length index
+            int bytesRead = fileOperations.read(messageArray, initialMessage.length, length);
+
+            String dataInFile = new String(messageArray, 0, initialMessage.length + bytesRead, StandardCharsets.UTF_8);
+            assertThat(dataInFile).isEqualTo("Hello! This is data ");
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+    }
+
+
+    @Test
+    void validateBytesOfDataReadWhileReadingItFromOffset() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            byte[] byteData = new byte[50];
+            int bytesRead = fileOperations.read(byteData, 0, 10);
+
+            assertThat(bytesRead).isEqualTo(10);
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+    @Test
+    void validateAllBytesReadingData() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            byte[] bytes = fileOperations.readAllBytes();
+            assertThat(fileOperations.getBytesRead()).isEqualTo(bytes.length);
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+
+    @Test
+    void validateAllBytesReadOperationCount() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            fileOperations.read();
+            fileOperations.read();
+            fileOperations.readAllBytes();
+            assertThat(fileOperations.getReadOperationCount()).isEqualTo(3);
+
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+    }
+
+
+    @Test
+    void validateReadNBytesReadOperationCount() {
+
+        try (Paasio fileOperations = new Paasio(this.dataInputStream, this.dataOutputStream)) {
+
+            fileOperations.readNBytes(5);
+            fileOperations.readNBytes(5);
+
+            assertThat(fileOperations.getReadOperationCount()).isEqualTo(2);
+            System.out.println("haha ");
+            System.out.println(fileOperations.getReadOperationCount());
```

Comment:
Please remove all `System.out.println`.

```suggestion
            System.out.println("haha ");
            System.out.println(fileOperations.getReadOperationCount());
```